src/lib/registration/controllers/package_search.rb

```diff
@@ -0,0 +1,189 @@
+# Copyright (c) [2020] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "registration/package_search"
+
+Yast.import "Popup"
+Yast.import "HTML"
+
+module Registration
+  module Controllers
+    # Implements the actions and keeps the state for the package search feature
+    class PackageSearch
+      include Yast::I18n
+
+      # @return [Array<RemotePackage>] List of selected packages
+      attr_reader :selected_packages
+
+      # Constructor
+      def initialize
+        textdomain "registration"
+        @selected_packages = []
+      end
+
+      # Returns the list of the current search
+      #
+      # @return [Array<RemotePackage>] List of found packages
+      def packages
```

Comment:
I am lost here. What is difference to selected packages? This is all packages that are result of search even not selected?

Reply:
`#packages` is the list of found packages, and `#selected_packages` is the current selection of packages. Now that we agreed to move the `#packages` list in the UI it should not be so confusing.